nixui/graphics/navlist.py

```diff
@@ -20,6 +20,130 @@ def __new__(cls, statemodel, set_option_path_fn, option_path, option_type=None,
             return StaticAttrsOf(option_path, set_option_path_fn, selected)
 
 
+class OptionListItemDelegate(QtWidgets.QStyledItemDelegate):
+    padding = 5
+
+    def paint(self, painter, option, index):
+        # ensure different background colors are applied for selected rows
+        viewOption = QtWidgets.QStyleOptionViewItem(option)
+        self.initStyleOption(viewOption, index)
+        QtWidgets.QStyledItemDelegate.paint(self, painter, viewOption, index)
+
+        data = index.data(QtCore.Qt.DisplayRole)
+
+        # draw left side icon if exists
+        if data.get('icon_path'):
+            icon_rect = QtCore.QRect(
+                option.rect.left() + self.padding,
+                option.rect.top() + self.padding,
+                option.rect.height() - self.padding * 2,
+                option.rect.height() - self.padding * 2,
+            )
+            icon_pixmap = icon.get_pixmap(data['icon_path'])
+            painter.drawPixmap(icon_rect, icon_pixmap)
+
+        if data.get('status_circle_color'):
+            status_circle_rect = QtCore.QRectF(
+                option.rect.right() - option.rect.height() * 0.5 - self.padding,
+                option.rect.top() + option.rect.height() * 0.5 - self.padding,
+                option.rect.height() * 0.25,
+                option.rect.height() * 0.25,
+            )
+            circle_outline_width = 2
+            painter.setPen(QtGui.QPen(QtCore.Qt.black, circle_outline_width, QtCore.Qt.SolidLine))
+            painter.setBrush(QtGui.QBrush(data['status_circle_color'], QtCore.Qt.SolidPattern))
+            painter.drawEllipse(status_circle_rect)
+
+        # get text ready to draw
+        text_left_offset = option.rect.height() + self.padding * 2 if data.get('icon_path') else self.padding * 2
+        text_right_offset = option.rect.height() + self.padding * 2  # padding for space from status circle
+        text_rect = QtCore.QRect(
+            option.rect.left() + text_left_offset + self.padding,
+            option.rect.top() + self.padding,
+            option.rect.width() - text_left_offset - text_right_offset - self.padding * 2,
+            option.rect.height() - self.padding * 2,
+        )
+        text = data['text'] + (f" ({data['child_count']})" if data.get('child_count') else "")
+
+        # draw subtext
+        if data.get('extra_text'):
+            painter.save()
+            extra_text_rect = QtCore.QRect(text_rect)
+            extra_text_rect.setY(text_rect.y() + option.rect.height() / 3)
+            text_rect.setY(text_rect.y() - option.rect.height() / 3)  # adjust the base text upwards
+            extra_text_font = QtGui.QFont()
+            extra_text_font.setItalic(True)
```

Comment:
hard to read

Reply:
This PR bundles a lot of changes stacked together. This is part of https://github.com/nix-gui/nix-gui/pull/187/files

I'll change it there.